instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -244,4 +249,5 @@ async def wrapped_send(message):
 
                 await self.app(scope, wrapped_receive, wrapped_send)
         finally:
-            context.detach(token)
+            if ctx is not None:
+                context.detach(token)
```

Comment:
This works now but it looks like something that could easily break in future with the value of `ctx` determining whether `token` is defined or not. Why not have a none-check for `token` directly instead?